### ISSUE TITLE
[Megatron] apply_liger_kernel_to_megatron(cross_entropy=True) does not reach Megatron's CE call sites

### ISSUE THREAD

**Issue by Rapisurazurite** [linkedin/Liger-Kernel#1485]
### 🐛 Describe the bug

`apply_liger_kernel_to_megatron(cross_entropy=True)` replaces `fused_vocab_parallel_cross_entropy` and `vocab_parallel_cross_entropy` only on their defining modules ([L199](https://github.com/linkedin/Liger-Kernel/blob/09f7febf1a4bfbb87c6b36e6a80cab864e33ee6e/src/liger_kernel/megatron/monkey_patch.py#L199), [L262](https://github.com/linkedin/Liger-Kernel/blob/09f7febf1a4bfbb87c6b36e6a80cab864e33ee6e/src/liger_kernel/megatron/monkey_patch.py#L262)). Megatron calls copies of these functions that its modules bound at import time. When Megatron model code is imported before the patch, which is the usual order and the one used in `examples/megatron/run_mode1_monkey_patch.py`, the loss keeps using Megatron's kernel. Nothing errors, and the INFO log says "Patched ...".

- megatron-core 0.18 / 0.19: `language_module.py` does `from megatron.core.fusions.fused_cross_entropy import fused_vocab_parallel_cross_entropy` ([L17](https://github.com/NVIDIA/Megatron-LM/blob/571370c829ca768fe37244f4e2e7f28d8accc4ab/megatron/core/models/common/language_module/language_module.py#L17)). The unfused branch calls the `megatron.core.tensor_parallel` package re-export ([L183](https://github.com/NVIDIA/Megatron-LM/blob/571370c829ca768fe37244f4e2e7f28d8accc4ab/megatron/core/models/common/language_module/language_module.py#L183)), which is bound before the `cross_entropy` submodule can be patched, so the unfused path never reaches Liger, whatever the order.
- Megatron-LM main (`NVIDIA/Megatron-LM#6766`, not released yet): `backends.py` binds the fused function as `_fused_ce` ([L19-L21](https://github.com/NVIDIA/Megatron-LM/blob/ec806b20a0a976924c1668486c3554ffb90c000e/megatron/core/models/backends.py#L19-L21)). The fused path depends on import order; the unfused path works.

The CE tests in `test/megatron/test_monkey_patch.py` only assert that the module attributes were replaced ([L369-L399](https://github.com/linkedin/Liger-Kernel/blob/09f7febf1a4bfbb87c6b36e6a80cab864e33ee6e/test/megatron/test_monkey_patch.py#L369-L399)), so this is not caught.

### Reproduce

A 1-layer `GPTModel` runs one forward and backward pass; the script counts calls into `LigerMegatronCrossEntropy.forward`.

```python
"""python repro.py import-first|patch-first  (single GPU, TP=1)"""

import sys

import torch
import torch.distributed as dist

order = sys.argv[1]
if order == "import-first":
    from megatron.core.models.gpt.gpt_model import GPTModel  # noqa: F401

from liger_kernel.megatron import apply_liger_kernel_to_megatron

apply_liger_kernel_to_megatron(rms_norm=False, cross_entropy=True)

import megatron.core  # noqa: E402
from liger_kernel.megatron import cross_entropy as liger_megatron_ce  # noqa: E402
from megatron.core import parallel_state  # noqa: E402
from megatron.core.models.gpt.gpt_layer_specs import get_gpt_layer_local_spec  # noqa: E402
from megatron.core.models.gpt.gpt_model import GPTModel  # noqa: E402
from megatron.core.tensor_parallel.random import model_parallel_cuda_manual_seed  # noqa: E402
from megatron.core.transformer.transformer_config import TransformerConfig  # noqa: E402

calls = [0]
forward = liger_megatron_ce.LigerMegatronCrossEntropy.forward


def counting_forward(self, *args, **kwargs):
    calls[0] += 1
    return forward(self, *args, **kwargs)


liger_megatron_ce.LigerMegatronCrossEntropy.forward = counting_forward

torch.cuda.set_device(0)
dist.init_process_group("nccl", rank=0, world_size=1, init_method="tcp://127.0.0.1:29615")
parallel_state.initialize_model_parallel(tensor_model_parallel_size=1, pipeline_model_parallel_size=1)
model_parallel_cuda_manual_seed(0)

vocab, seq, batch = 1024, 32, 2
tokens = torch.randint(0, vocab, (batch, seq), device="cuda")
labels = torch.randint(0, vocab, (batch, seq), device="cuda")
position_ids = torch.arange(seq, device="cuda").unsqueeze(0).expand(batch, -1)
attention_mask = torch.ones(batch, 1, seq, seq, dtype=torch.bool, device="cuda").triu(1)
print(f"megatron-core {megatron.core.__version__}, order={order}")
for fusion in (True, False):
    config = TransformerConfig(
        num_layers=1,
        hidden_size=64,
        num_attention_heads=4,
        use_cpu_initialization=True,
        cross_entropy_loss_fusion=fusion,
        cross_entropy_fusion_impl="native",
    )
    model = GPTModel(
        config=config, transformer_layer_spec=get_gpt_layer_local_spec(), vocab_size=vocab, max_sequence_length=seq
    ).cuda()
    calls[0] = 0
    loss = model(tokens, position_ids, attention_mask, labels=labels)
    loss.mean().backward()
    print(f"  cross_entropy_loss_fusion={fusion}: liger calls = {calls[0]}")
dist.destroy_process_group()
```

Output with megatron-core 0.18.2 and liger-kernel 0.8.3:

```
megatron-core 0.18.2, order=import-first
  cross_entropy_loss_fusion=True: liger calls = 0
  cross_entropy_loss_fusion=False: liger calls = 0
megatron-core 0.18.2, order=patch-first
  cross_entropy_loss_fusion=True: liger calls = 1
  cross_entropy_loss_fusion=False: liger calls = 0
```

Expected: 1 in every case. On Megatron-LM main the same script gives 0 / 1 for import-first and 1 / 1 for patch-first (fused / unfused).

### Versions

- liger-kernel 0.8.3 (the patch code is unchanged on main `09f7feb`)
- megatron-core 0.18.2 (0.19.2 has the same call sites); Megatron-LM main `c9d5abb`
- torch 2.13.0+cu130, triton 3.7.1, CUDA 13.0, Python 3.12, 1x NVIDIA GPU
